lib/rules/declaration-block-no-duplicate-properties/index.js

```diff
@@ -88,97 +88,84 @@ const rule = (primary, secondaryOptions, context) => {
-									decl.remove();
-								} else {
-									removePreviousDuplicate(decls, lowerProp);
-								}
-
-								return;
-							}
+				if (indexDuplicate === -1) {
+					decls.push(decl);
+				}
+
+				const duplicateDecl = decls[indexDuplicate];
+
+				if (!duplicateDecl) {
+					return;
+				}
+
+				const duplicateValue = duplicateDecl.value || '';
+				const duplicateImportant = duplicateDecl.important || false;
+				const duplicateIsMoreImportant = !important && duplicateImportant;
+				const duplicatesAreConsecutive = indexDuplicate === decls.length - 1;
 
-							report({
+				if (ignoreDiffValues || ignorePrefixlessSameValues) {
+					if (!duplicatesAreConsecutive) {
+						if (!context.fix) {
+							return report({
 								message: messages.rejected(prop),
 								node: decl,
 								result,
 								ruleName,
 								word: prop,
 							});
-
-							return;
 						}
 
-						if (ignorePrefixlessSameValues) {
-							// fails if values of consecutive, unprefixed duplicates are equal
-							if (vendor.unprefixed(value) !== vendor.unprefixed(duplicateValue)) {
-								if (context.fix) {
-									if (!important && duplicateImportant) {
-										decl.remove();
-									} else {
-										removePreviousDuplicate(decls, lowerProp);
-									}
+						if (duplicateIsMoreImportant) {
+							return decl.remove();
+						}
 
-									return;
-								}
+						return duplicateDecl.remove();
+					}
 
-								report({
+					if (ignorePrefixlessSameValues) {
+						// fails if values of consecutive, unprefixed duplicates are equal
+						if (vendor.unprefixed(value) !== vendor.unprefixed(duplicateValue)) {
+							if (!context.fix) {
+								return report({
 									message: messages.rejected(prop),
 									node: decl,
 									result,
 									ruleName,
 									word: prop,
 								});
-
-								return;
 							}
-						}
-
-						// fails if values of consecutive duplicates are equal
-						if (value === duplicateValue) {
-							if (context.fix) {
-								removePreviousDuplicate(decls, lowerProp);
 
-								return;
+							if (duplicateIsMoreImportant) {
+								return decl.remove();
 							}
 
-							report({
-								message: messages.rejected(prop),
-								node: decl,
-								result,
-								ruleName,
-								word: prop,
-							});
-
-							return;
+							return duplicateDecl.remove();
 						}
-
-						return;
 					}
 
-					if (ignoreDuplicates && indexDuplicate === decls.length - 1) {
+					if (value !== duplicateValue) {
 						return;
 					}
 
 					if (context.fix) {
-						if (!important && duplicateImportant) {
-							decl.remove();
-						} else {
-							removePreviousDuplicate(decls, lowerProp);
-						}
-
-						return;
+						return duplicateDecl.remove();
 					}
 
-					report({
+					return report({
+						message: messages.rejected(prop),
+						node: decl,
+						result,
+						ruleName,
+						word: prop,
+					});
+				}
+
+				if (ignoreDuplicates && duplicatesAreConsecutive) {
```

Comment:
question

Do we have a test for
ignoreDuplicates === true
duplicatesAreConsecutive === false
content.fix === true
duplicateIsMoreImportant === false
?
i.e. in that case should we reach l181? Is this setup possible?